src/clj/puppetlabs/services/master/master_core.clj

```diff
@@ -916,13 +926,50 @@
     ;; default to itself
     mount))
 
+(defn read-bolt-project-config
+  [project-dir]
+  (let [config-path (str project-dir "/bolt-project.yaml")]
+    (if (fs/file? config-path)
+     (yaml/parse-string (slurp config-path)))))
+
+(def default-project-modulepath
+  ["modules" "site-modules" "site"])
+
+(defn parse-modulepath
+  "The modulepath for a bolt project can either be an array of paths or a string
+  with a path separator."
+  [modulepath]
+  (if (string? modulepath)
+    (str/split modulepath #":")
+    modulepath))
+
+(defn get-project-modulepath
+  "Given a map representing the bolt project configuration, return the
+  modulepath. If the modulepath is not defined or the configuration map passed
+  is nil then return the default modulepath."
+  [project-config]
+  (let [has-modules-key? (contains? project-config :modules)]
+    (if-let [modulepath (parse-modulepath (get project-config :modulepath))]
+      (if has-modules-key?
+        (concat modulepath [".modules"])
```

Comment:
There's a lot I didn't understand about Alex's email, so I might be wrong, but I thought ".modules" gets appended to the specified module path, so shouldn't this be something like `(map #(str % ".modules") modulepath)`?
Or an alternative interpretation.. `.modules` is the default modulepath, so we should put that first instead of last (`(concat [".modules"] modulepath)`)?

Reply:
I think the way I did it is right, based on reading the code in bolt.

Reply:
puppetlabs/bolt/pull/2189